### DISCUSSION TITLE
Projections after slicing including flow bins

### DISCUSSION THREAD

**Discussion by NJManganelli** [scikit-hep/hist#384] [Q&A] [answered]
I have a large-D histogram which I have sliced, and I've intentionally chosen one slice in "nbtag" which is empty.

```
bkg = test[['ST', 'ttH', 'ttVJets', 'ttrare', 'Diboson', 'WJets', 'DY', 'ttbb', 'ttjj'],
               [model+'_'+mode], 0j:2j, 4j:5j, ...].project("nbtag", "njet", "dataset")
```
That gives 
```
Regular(2, 0, 2, underflow=False, name='nbtag', label='𝑁𝑏𝑡𝑎𝑔')
Regular(1, 4, 5, name='njet', label='𝑁𝑗𝑒𝑡')
StrCategory(['ST', 'ttH', 'ttVJets', 'ttrare', 'Diboson', 'WJets', 'DY', 'ttbb', 'ttjj'], growth=True, name='dataset', label='dataset')
Weight() Σ=WeightedSum(value=0, variance=0) (WeightedSum(value=6233.93, variance=532.169) with flow)
```
As expected!

Now I project down another level in nbtag:
```
bkg.project("nbtag", "dataset")
Regular(2, 0, 2, underflow=False, name='nbtag', label='𝑁𝑏𝑡𝑎𝑔')
StrCategory(['ST', 'ttH', 'ttVJets', 'ttrare', 'Diboson', 'WJets', 'DY', 'ttbb', 'ttjj'], growth=True, name='dataset', label='dataset')
Weight() Σ=WeightedSum(value=0, variance=0) (WeightedSum(value=6233.93, variance=532.169) with flow)
```
Yep! Still what I want

Now in njet
```
bkg.project("njet", "dataset")
Regular(1, 4, 5, name='njet', label='𝑁𝑗𝑒𝑡')
StrCategory(['ST', 'ttH', 'ttVJets', 'ttrare', 'Diboson', 'WJets', 'DY', 'ttbb', 'ttjj'], growth=True, name='dataset', label='dataset')
Weight() Σ=WeightedSum(value=2396.36, variance=314.223) (WeightedSum(value=6233.93, variance=532.169) with flow)
```
Not quite what I wanted, nor expected! I sliced the histogram afterall, but the flow from nbtag has been included back in I suppose, while the slice on njet is still applicable?

So the question is, how can I do what I intended (e.g. project down to the njet/dataset axes, but still stick to the non-flow slice in nbtag from defining 'bkg')? If it's including 'flow' along this axis that had been sliced, what about other axes when I projected down from even higher dimensions?

**Comment by nsmith-** [comment, marked as answer]
Does `bkg[0j:2j:sum, ...]` return the expected value (as opposed to `bkg.project("njet", "dataset")`)?
In general, I think the `project` function needs an option to decide whether or not to include flow bins.

**Reply by NJManganelli** [reply]
Forgot to add this to the discussion, yes, calling sum in the slice appeared to give the expected results.
I agree, the project function could use this for the interface.
